src/components/ConsentPicker.vue

```diff
@@ -0,0 +1,486 @@
+            <div class="flex flex-row w-full">
+              <div style="width: 80%">
+                <p class="m-0">
+                  <span class="font-bold">Name: </span>{{ result.assent[0]?.fileName }} <br />
+                  <span class="font-bold">Current Commit: </span>{{ result.assent[0]?.currentCommit }}
+                  <br />
+                  <span class="font-bold">GitHub Org: </span>{{ result.assent?.gitHubOrg }} <br />
+                  <span class="font-bold">GitHub Repository: </span>{{ result.assent[0]?.gitHubRepository }}
+                  <br />
+                  <span class="font-bold">Last Updated: </span>{{ result.assent[0]?.lastUpdated }} <br />
+                </p>
+              </div>
+              <div class="flex align-items-center justify-content-center">
+                <PvButton
+                  class="border-circle w-6rem h-6rem m-2 surface-hover text-primary border-none font-bold flex align-items-center justify-content-center hover:text-100 hover:bg-primary"
+                  label="Show Assent"
+                  @click="seeConsent(result.assent[0])"
+                />
+              </div>
+            </div>
+          </PvFieldset>
+          <div v-else>
+            <p>No assent available.</p>
+          </div>
+        </div>
+      </div>
+    </div>
+  </PvPanel>
+  <PvDialog
+    v-if="showConsent"
+    v-model:visible="showConsent"
+    :draggable="false"
+    modal
+    :close-on-escape="false"
+    :style="{ width: '65vw' }"
+    :breakpoints="{ '1199px': '85vw', '575px': '95vw' }"
+  >
+    <div v-html="confirmText"></div>
+  </PvDialog>
+</template>
+
+<script setup>
+import { ref, onMounted, watch, computed } from 'vue';
+import { fetchLegalDocs } from '@/helpers/query/legal';
+import { useQuery } from '@tanstack/vue-query';
+import { useAuthStore } from '@/store/auth';
+import { marked } from 'marked';
+import _lowerCase from 'lodash/lowerCase';
+import _mapValues from 'lodash/mapValues';
+import _forEach from 'lodash/forEach';
+
+const props = defineProps({
+  legal: { type: Object, required: false, default: null },
+});
+
+const defaultParams = [
+  {
+    name: 'Mouse and click',
+    icon: 'pi pi-check',
+  },
+  {
+    name: 'Questionnaire responses',
+    icon: 'pi pi-list',
+  },
+  {
+    name: 'Scrolling behavior',
+    icon: 'pi pi-arrows-v',
+  },
+  {
+    name: 'Button presses',
+    icon: 'pi pi-plus-circle',
+  },
+];
+
+const specialParams = [
+  {
+    name: 'eye-tracking',
+  },
+  {
+    name: 'video recording',
+  },
+  {
+    name: 'audio recording',
+  },
+];
+
+const initialized = ref(false);
+const showConsent = ref(false);
+const consentVersion = ref('');
+const confirmText = ref('');
+const paramCheckboxData = ref(false);
+const specialParam = ref(false);
+const amount = ref('');
+const expectedTime = ref('');
+const helpMeSelect = ref(null);
+let selectedConsent = ref(null);
+let selectedAssent = ref(null);
+const knowWhatIWant = ref(false);
```

Comment:
Can I suggest that this variable name be renamed to `userDrivenFlow` (or something similar)? Furthermore, do you think renaming `helpMeSelect` to `userAssistedFlow` would help with readability and clarity?

Reply:
sure!